src/Core/GraphQl/Serializer/SerializerContextBuilder.php

```diff
@@ -28,38 +29,45 @@
  */
 final class SerializerContextBuilder implements SerializerContextBuilderInterface
 {
-    private $resourceMetadataFactory;
+    private $resourceMetadataCollectionFactory;
     private $nameConverter;
 
-    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, ?NameConverterInterface $nameConverter)
+    public function __construct(ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory, ?NameConverterInterface $nameConverter)
     {
-        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->resourceMetadataCollectionFactory = $resourceMetadataCollectionFactory;
         $this->nameConverter = $nameConverter;
     }
 
     public function create(?string $resourceClass, string $operationName, array $resolverContext, bool $normalization): array
     {
-        $resourceMetadata = $resourceClass ? $this->resourceMetadataFactory->create($resourceClass) : null;
-
-        $context = [
-            'resource_class' => $resourceClass,
-            'graphql_operation_name' => $operationName,
-        ];
+        $context = ['operation_name' => $operationName, 'graphql_operation_name' => $operationName];
```

Comment:
Why setting `operation_name`?

Reply:
Because it may be used by the IRI Converter.